src/prefect/cli/work_pool.py

```diff
@@ -380,6 +380,58 @@ async def update(
         exit_with_success(f"Updated work pool {name!r}")
 
 
+@work_pool_app.command()
+async def provision_infra(
+    name: str = typer.Argument(
+        ..., help="The name of the work pool to provision infrastructure for."
+    ),
+):
+    """
+    Provision infrastructure for a work pool.
+
+    \b
+    Examples:
+        $ prefect work-pool provision-infrastructure "my-pool"
+
+    """
+    if not name:
+        exit_with_error("Work pool name cannot be empty.")
```

Comment:
I think typer will enforce this for you because you typed `name` as `str` without a default value.

Reply:
Makes sense!

Reply:
I've removed this in [8a8ddb4](https://github.com/PrefectHQ/prefect/pull/11341/commits/8a8ddb4725fb547e5858357370906600eb9c4afe)